hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/CapacitySchedulerConfiguration.java

```diff
@@ -1560,82 +1545,82 @@ void setWorkflowPriorityMappings(
 
   public boolean isReservable(String queue) {
     boolean isReservable =
-        getBoolean(getQueuePrefix(queue) + IS_RESERVABLE, false);
+        getBoolean(getQueuePrefix(new QueuePath(queue)) + IS_RESERVABLE, false);
     return isReservable;
   }
 
   public void setReservable(String queue, boolean isReservable) {
-    setBoolean(getQueuePrefix(queue) + IS_RESERVABLE, isReservable);
+    setBoolean(getQueuePrefix(new QueuePath(queue)) + IS_RESERVABLE, isReservable);
     LOG.debug("here setReservableQueue: queuePrefix={}, isReservableQueue={}",
-        getQueuePrefix(queue), isReservable(queue));
+        getQueuePrefix(new QueuePath(queue)), isReservable(queue));
   }
 
   @Override
   public long getReservationWindow(String queue) {
     long reservationWindow =
-        getLong(getQueuePrefix(queue) + RESERVATION_WINDOW,
+        getLong(getQueuePrefix(new QueuePath(queue)) + RESERVATION_WINDOW,
             DEFAULT_RESERVATION_WINDOW);
     return reservationWindow;
   }
 
   @Override
   public float getAverageCapacity(String queue) {
     float avgCapacity =
-        getFloat(getQueuePrefix(queue) + AVERAGE_CAPACITY,
+        getFloat(getQueuePrefix(new QueuePath(queue)) + AVERAGE_CAPACITY,
             MAXIMUM_CAPACITY_VALUE);
     return avgCapacity;
   }
 
   @Override
   public float getInstantaneousMaxCapacity(String queue) {
     float instMaxCapacity =
-        getFloat(getQueuePrefix(queue) + INSTANTANEOUS_MAX_CAPACITY,
+        getFloat(getQueuePrefix(new QueuePath(queue)) + INSTANTANEOUS_MAX_CAPACITY,
             MAXIMUM_CAPACITY_VALUE);
     return instMaxCapacity;
   }
 
-  public void setInstantaneousMaxCapacity(String queue, float instMaxCapacity) {
+  public void setInstantaneousMaxCapacity(QueuePath queue, float instMaxCapacity) {
     setFloat(getQueuePrefix(queue) + INSTANTANEOUS_MAX_CAPACITY,
         instMaxCapacity);
   }
 
-  public void setReservationWindow(String queue, long reservationWindow) {
+  public void setReservationWindow(QueuePath queue, long reservationWindow) {
     setLong(getQueuePrefix(queue) + RESERVATION_WINDOW, reservationWindow);
   }
 
-  public void setAverageCapacity(String queue, float avgCapacity) {
+  public void setAverageCapacity(QueuePath queue, float avgCapacity) {
     setFloat(getQueuePrefix(queue) + AVERAGE_CAPACITY, avgCapacity);
   }
 
   @Override
   public String getReservationAdmissionPolicy(String queue) {
```

Comment:
getReservationAdmissionPolicy / setReservationAdmissionPolicy has different API. The getter works with queue name (path?) but the setter works with QueuePath object. I think this should be streamlined.

Reply:
I noticed the same pattern for setReservationAgent / getReservationAgent as well. Please revisit these and try to make getters / setters consistent in terms of parameter types.

Reply:
Thanks for pointing it out, I continued the replacing in the latest commits to make it consistent